### ISSUE TITLE
Accessibility issues make the website less accessible to people with visual disabilities and people using screen readers

### ISSUE THREAD

**Issue by pangolingo** [ApprenticeshipStandardsDotOrg/ApprenticeshipStandardsDotOrg#1088]
# 1. Homepage YouTube video embed is missing title attributes

Page:
https://apprenticeshipstandards.org/

Description:
iFrames must have an accessible name - this iFrame has no title tag. Some screen readers may not know how to convey this element's meaning to the user.

Severity: low. The iFrame contents have a title tag, and the YouTube video is directly below a screenreader-only headline that describes it

How to fix:
Add a title attribute to the iframe with the title of the video. Consider using YouTube's default iFrame embed code which includes a title tag.

# 2. Blue subheads have insufficient contrast

Page:
https://apprenticeshipstandards.org/
and others

Description:
Blue subheads on a light blue background have insufficient contrast. Some people with visual disabilities may find them hard to read. WCAG guidelines recommend a minimum contrast ratio of 4.5:1. The contrast ratio for these colors is only 2.81:1.
Colors (foreground: #5BA5A5, background: #F8FFFF)

[Checked with WebAIM's contrast checker.](https://webaim.org/resources/contrastchecker/)

![Image](https://github.com/user-attachments/assets/da6f68d7-bffd-4630-be4f-d17f9b40db7a)

Severity: medium. This violates WCAG 1.4.3, level AA

How to fix:
Use a darker color for the headlines.

# 3. `<a>` tags should not be nested inside `<button>` tags

Page:
https://apprenticeshipstandards.org/occupation_standards
This occurs in the search result badges.
This also occurs on the apprenticeship page that you access by clicking on a search result

Description:
The badges on some pages have an `<a>` tag nested inside a `<button>` tag. Not only is this invalid HTML, it also makes it hard for screenreaders to properly announce the button or link.

Severity: low.

How to fix:
These badges appear to function as a link. Remove the outer `<button>` tag and move its classes to the `<a>` tag instead.

# 4. "State" search filter dropdown is not labeled in an accessible way

Page: https://apprenticeshipstandards.org/occupation_standards?q=15-1132.00
and other pages with search filters

Description:
If you go to the search module and press "expand filters", you see more filter options include a "state" dropdown. This dropdown is not accessibly labeled. It does have a label associated, but that label is hidden. Fortunately the Voiceover screenreader still announces the name. But other screenreaders may not.

Severity: low to medium

How to fix:
Change the label's "hidden" class to "sr-only". This should reveal the label to screenreaders while keeping it visually hidden.

# 5. Heading tags should not be empty

Page:
https://apprenticeshipstandards.org/occupation_standards?q=15-1132.00
and others with search results

Description:
There is an empty `<h2>` tag inside each search result card component. Empty headline tags will be confusing to unsighted users who are using a screenreader. These empty tag are between the card's `<h1>` tag with the name of the apprenticeship and the badges. They only occur on cards without an associated company - I believe they are meant to show the company name.

Severity: low - these are not a big obstacle to users

How to fix:
When there is no company associated with an apprenticeship, hide the entire `<h2>` tag from the search result card, instead of just hiding the text inside the `<h2>`.

# 6. All pages must have an `<h1>` tag

Page: https://apprenticeshipstandards.org/occupation_standards/1051f9f5-508b-4329-8a09-3ae447b969b7

Description:
This type of page is missing a level-1 heading tag that describes the page. This will make it more difficult for screenreader users to determine what the page is about, and will make it difficult for them to jump to the main contents of the page.

Severity: low - the page has a title tag which will convey this information

How to fix:
Change the "Software Engineer" `<h3>` headline to an `<h1>`.

# 7. Pages should not have multiple "main" landmarks

Page:
https://apprenticeshipstandards.org/states

Description:
Screenreader users use `<main>` landmarks to navigate around the page and jump to the primary page content. This page has both a `<main>` element and a `<section role="main">` element. Additionally, the `<main>` tag wraps the header as well as the actual main content. 

Severity: low - there are other ways to navigate besides using landmarks

How to fix:
Change the wrapper `<main>` to a `<div>`, and change `<section role="main">` to `<main>`. Test a variety of pages to ensure that they all have a `<main>` tag that only wraps the primary page contents, not including headers and footers. Wrap the dark green header in a `<header>` tag, giving it the "banner" landmark role. 

# 8. Test the API documentation pages for accessibility issues

The API docs at https://developers.apprenticeshipstandards.org/#introduction appear to have some issues with contrast, zooming, and keyboard focus. They should probably be tested for accessibility to make sure everyone is able to learn how to use the API.
